### PR TITLE
Coerce mod agnostic

### PR DESCRIPTION
**Pull request by joanrue**
Coerce() function can now handle cupy and dask arrays.

### CODE REVIEW

File: src/pycsou/abc/solver.py [hunk 1]
```diff
@@ -566,6 +566,7 @@ def _m_persist(self):
 
     class _Worker(threading.Thread):
         def __init__(self, solver: "Solver"):
+            super().__init__()
```
Comment (SepandKashani) [inline src/pycsou/abc/solver.py:569]: Not really needed, but ok.

File: src/pycsou/runtime/__init__.py [hunk 2]
```diff
@@ -138,13 +139,21 @@ def coerce(x):
         Fails if operation is impossible or unsafe. (I.e. casting complex-valued data.)
     """
     dtype = getPrecision().value
+    try:
```
Comment (SepandKashani) [inline src/pycsou/runtime/__init__.py]: I don't quite agree with all the steps here. Let's talk about this in the afternoon.

**Review by SepandKashani — COMMENTED** [review]
Have yet to look at `cg.py`.

File: src/pycsou/runtime/__init__.py [hunk 3]
```diff
@@ -7,6 +7,7 @@
 
 import numpy as np
 
+import pycsou.util as pycu
```
Comment (SepandKashani) [inline src/pycsou/runtime/__init__.py]: Not required anymore.

File: src/pycsou/abc/solver.py (hunk 1 above)
Comment (SepandKashani) [inline src/pycsou/abc/solver.py]: ok, based on our discussion.

**Review by SepandKashani — CHANGES_REQUESTED** [review]
Some minor things to fix. (See comments.)
For `cg.py`, I suggest removing it entirely from this pull request and adding it only to #15

**Comment by SepandKashani** [comment]
@joanrue Are you currently working on this?
Only `runtime/__init__.py` should be modified in this pull request. (No `xp` inference needed, just drop the `casting=` parameter and it should be good to go.

**Comment by joanrue** [comment]
@SepandKashani Sorry made a mistake before you explained me the stash (This should not happen again). I fixed the conflict and I think this branch is ready to be merged.

**Review by SepandKashani — APPROVED** [review]
Looks good.

**Pull request merged**